addon/state-for.js

```diff
@@ -3,42 +3,50 @@ import WeakMap from 'ember-weakmap/weak-map';
 
 var {
   computed,
-  assert
+  assert,
+  typeOf
 } = Ember;
 
 /*
- * Each key in this POJO is a name of a state with the value
- * for that key being a WeakMap.
- */
+* Each key in this POJO is a name of a state with its value being a WeakMap.
+*
+* {
+*    'my-state-name': WeakMapFoo,
+*    'my-other-name': WeakMapBar
+* }
+*/
 var weakMaps = {};
 
 /*
-* Returns a computed property if called like: stateFor('state-name', 'property-name') or
-* returns the WeakMap if called like stateFor('state-name'). The most common case will be
-* the former but the latter allows advanced options like:
+* StateFor will return a computed property that will contain the state 'bucket'
+* which is assoicated with the 'dependent-key-path' property. An optional third
+* argument can be passed to give a more granular division of state.
 *
-* stateFor('state-name').delete(this.get('property-name'));
-* stateFor('state-name').has(this.get('some-other-prop'));
-* stateFor('state-name').set(this.get('property-name'), {}); // Overrides the state
-* stateFor('state-name').get(this.get('property-name'));
+* stateFor('state-name', 'dependent-key-path');
+* stateFor('state-name', 'dependent-key-path',  {
+*    secondaryKeys: ['foo', 'bar']
```

Comment:
right now they are just namespaces. bound keys get weird if they are bound to non primitives as it was throwing guidFor off